lib/site_encrypt/phoenix.ex

```diff
@@ -28,21 +28,11 @@ defmodule SiteEncrypt.Phoenix do
   defp acme_spec(%{ca_url: url}, _endpoint) when is_binary(url), do: nil
 
   defp acme_spec(%{ca_url: {:local_acme_server, acme_server_config}} = config, endpoint) do
-    %{port: port, adapter: adapter} = acme_server_config
+    %{port: port, adapter: _adapter} = acme_server_config
     SiteEncrypt.Logger.log(config.log_level, "Running local ACME server at port #{port}")
 
-    AcmeServer.Standalone.child_spec(
-      adapter: acme_server_adapter_spec(adapter, port),
-      dns: dns(config, endpoint)
-    )
-  end
-
-  defp acme_server_adapter_spec(Plug.Adapters.Cowboy, port),
-    do: {Plug.Adapters.Cowboy, scheme: :http, options: [port: port, acceptors: 1]}
-
-  defp dns(config, endpoint) do
-    [config.domain | config.extra_domains]
-    |> Enum.map(&{&1, fn -> "localhost:#{endpoint.config(:http) |> Keyword.fetch!(:port)}" end})
-    |> Enum.into(%{})
+    config
+    |> Map.put(:dns, AcmeEx.Endpoint.dns(endpoint, config))
+    |> AcmeEx.Standalone.child_spec()
```

Comment:
Here's the main swap out; based on the library

https://github.com/aforward/acme_ex